Quaver.Shared/Screens/Gameplay/Rulesets/Keys/Playfield/GameplayPlayfieldKeys.cs

```diff
@@ -108,28 +98,171 @@ public GameplayPlayfieldKeys(GameplayScreen screen, GameplayRulesetKeys ruleset)
             Screen = screen;
             Ruleset = ruleset;
             Container = new Container();
+            DetermineScrollDirections();
+            SetReceptorPositions();
+            ApplyHitPositionsOffset();
+            CreateElementContainers();
+        }
 
+        /// <summary>
+        ///     Create Foreground and Background Containers, as well as the Stage.
+        /// </summary>
+        private void CreateElementContainers()
+        {
             // Create background container
             BackgroundContainer = new Container
             {
                 Parent = Container,
                 Size = new ScalableVector2(Width, WindowManager.Height),
-                Alignment = Alignment.TopCenter,
-                X = SkinManager.Skin.Keys[screen.Map.Mode].ColumnAlignment
+                Alignment = Alignment.TopCenter
             };
 
             // Create the foreground container.
             ForegroundContainer = new Container
             {
                 Parent = Container,
                 Size = new ScalableVector2(Width, WindowManager.Height),
-                Alignment = Alignment.TopCenter,
-                X = BackgroundContainer.X
+                Alignment = Alignment.TopCenter
             };
 
             Stage = new GameplayPlayfieldKeysStage(Screen, this);
         }
 
+        /// <summary>
+        ///     Determine the Scroll Directions for each Lane.
+        /// </summary>
+        private void DetermineScrollDirections()
+        {
+            switch (Ruleset.Map.Mode)
+            {
+                case GameMode.Keys4:
+                    switch (ConfigManager.ScrollDirection4K.Value)
```

Comment:
Not a fan of the nested switches. Break it up into separate methods if u have to.